### DISCUSSION TITLE
Web3Modal v2 alpha discussion

### DISCUSSION THREAD

**Discussion by xzilja** [WalletConnect/walletconnect-monorepo#1549] [Developer Support]
As you may have noticed we recently released and are continuously working on new version of web3modal. In spirit of making it one of the best experiences when starting web3 projects, this discussion will serve as a gateway for community to suggest their ideas, report bugs and ask questions during alpha development.

**Comment by Crytogenz** [comment]
Really loving the new Web3Modal, there is only one thing I was wondering when will you be adding Infinity Wallet to the list? Most users that use my DApp use Infinity Wallet on desktop and I think its like the top most used desktop wallet after MetaMask.

I was looking to add support for this web3modal as it looks mint and noticed it doesn't have such support yet. At the moment we currently support Infinity Wallet via their direct connector here https://www.npmjs.com/package/@infinitywallet/infinity-connector, it would be great if you could add support for it so then we can 100% integrate this web3 modal.
![image](https://i.imgur.com/n6GWRFf.png)

Also something I would like to also recommend is that when on desktop and you click get a Wallet its lists only mobile wallets which is really bad. It needs to list desktop wallets MetaMask, Infinity Wallet, Ledger and etc...
![image](https://i.imgur.com/njGqlwB.png)

Also on the home screen of the modal when on desktop it should really show desktop wallets at the top and mobile wallets below, instead of how currently its mobile on top and desktop below.

Keep up the good work

**Reply by xzilja** [reply]
Thank you for feedback!

Infinity wallet is supported, but you have to click on "View All" button. Assuming their deep / universal links are up to date in our explorer, clicking on it should open desktop app and pass it wc connection uri. (same flow as ledger desktop)

At the moment we are just a bit busy focusing on stability and adding other features (chain switching ui / account ui). But more customisation options will be added soon for sure 👌 

<img width="438" alt="Screenshot 2022-09-26 at 15 55 46" src="https://user-images.githubusercontent.com/3154053/192281980-2128f0a9-4cb8-4e68-891f-eb7bbf307900.png">

**Reply by Crytogenz** [reply]
That is WalletConnect just, not their actual Infinity Wallet connecting via their Web3 Browser which is completely different and needs their connector supported to be able to connect with it.

**Reply by xzilja** [reply]
@Crytogenz small update on this, we are working on our new "universal" provider that will enable walletconnect v1 and v2 support in the modal and have also discussed custom connectors. So at some point we will allow you to create such connectors and prioritise them in the modal 👍
